pkg/controllers/binding/common.go

```diff
@@ -110,7 +103,7 @@ func ensureWork(
 				return err
 			}
 
-			if clonedWorkload.GetKind() == util.JobKind && jobHasCompletions {
+			if len(jobCompletions) > 0 {
 				// For a work queue Job that usually leaves .spec.completions unset, in that case, we skip setting this field.
 				// Refer to: https://kubernetes.io/docs/concepts/workloads/controllers/job/#parallel-jobs.
```

Comment:
```suggestion
			// Set allocated completions for Job only when the '.spec.completions' field not omitted from resource template.
			// For jobs running with a 'work queue' usually leaves '.spec.completions' unset, in that case we skip
			// setting this field as well.
			// Refer to: https://kubernetes.io/docs/concepts/workloads/controllers/job/#parallel-jobs.
			if len(jobCompletions) > 0 {
```

Reply:
I think the comment should be added above the `if condition`, I added some details.

Reply:
updated and squashed the commits.